apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -118,6 +158,30 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
                 }
                 outGPyramid[j].compute_root().gpu_tile(x, y, xi, yi, blockw, blockh);
             }
+        } else if (get_target().features_any_of({Target::HVX_64, Target::HVX_128
+}) && !float_point) {
+            const int vector_size = get_target().has_feature(Target::HVX_128) ? 128 : 64;
+            int vec_lanes_16 = vector_size / 2;
+            int vec_lanes_8  = vector_size;
+            Var yo;
+            output.reorder(c, x, y).split(y, yo, y, 64).parallel(yo).vectorize(x, 8);
+            gray.compute_root().parallel(y, 32).vectorize(x, vec_lanes_16);
+            for (int j = 1; j < 5; j++) {
+                inGPyramid[j]
+                    .compute_root().parallel(y, 32).vectorize(x, 8);
+                gPyramid[j]
+                    .compute_root().reorder_storage(x, k, y)
+                    .reorder(k, y).parallel(y, 8).vectorize(x, 8);
+                outGPyramid[j]
+                    .store_at(output, yo).compute_at(output, y).fold_storage(y, 8)
+                    .vectorize(x, 8);
+            }
+            outGPyramid[0].compute_at(output, y).vectorize(x, 8);
+            for (int j = 5; j < J; j++) {
+                inGPyramid[j].compute_root().hexagon();
+                gPyramid[j].compute_root().hexagon().parallel(k);
```

Comment:
This is going to generate a lot of RPC calls, which might have significant overhead compared to the runtime (these small pyramid levels are probably very fast compared to the RPC overhead).

This could be avoided by doing compute_at(consumer, Var::outermost()) instead of compute_root. The tradeoff is that will allocate a lot of memory on Hexagon, which can get "stuck" (Hexagon's heap grows, but never shrinks).